modules/discopower/lib/PowerIdPDisco.php

```diff
@@ -248,21 +248,9 @@ public function handleRequest()
         $preferredIdP = $this->getRecommendedIdP();
 
         $t = new \SimpleSAML\XHTML\Template($this->config, 'discopower:disco.tpl.php', 'disco');
-        $discoPowerTabs = [
-            'denmark' => \SimpleSAML\Locale\Translate::noop('{discopower:tabs:denmark}'),
-            'edugain' => \SimpleSAML\Locale\Translate::noop('{discopower:tabs:edugain}'),
-            'finland' => \SimpleSAML\Locale\Translate::noop('{discopower:tabs:finland}'),
-            'greece' => \SimpleSAML\Locale\Translate::noop('{discopower:tabs:greece}'),
-            'southafrica' => \SimpleSAML\Locale\Translate::noop('{discopower:tabs:southafrica}'),
-            'iceland' => \SimpleSAML\Locale\Translate::noop('{discopower:tabs:iceland}'),
-            'incommon' => \SimpleSAML\Locale\Translate::noop('{discopower:tabs:incommon}'),
-            'kalmar' => \SimpleSAML\Locale\Translate::noop('{discopower:tabs:kalmar}'),
-            'misc' => \SimpleSAML\Locale\Translate::noop('{discopower:tabs:misc}'),
-            'norway' => \SimpleSAML\Locale\Translate::noop('{discopower:tabs:norway}'),
-            'sweden' => \SimpleSAML\Locale\Translate::noop('{discopower:tabs:sweden}'),
-            'switzerland' => \SimpleSAML\Locale\Translate::noop('{discopower:tabs:switzerland}'),
-            'ukacessfederation' => \SimpleSAML\Locale\Translate::noop('{discopower:tabs:ukacessfederation}'),
-        ];
+        foreach($this->discoconfig->getValue('tabs', null) as $tab) {
```

Comment:
I'd use getArray here and use `[]` or `['misc']` as the default value.
Now this code breaks if the setting is not there, because we cannot loop over `null`